### PR TITLE
osbuild2: support 'install' command in the modprobe stage and rework data validation

### PR DESCRIPTION
**Pull request by thozza**
- osbuild2: support 'install' command in the modprobe stage
  - Related to osbuild/osbuild#867
- osbuild2/modprobe: extend and rework stage options validation
  -  Related to issue #1785.

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by thozza** [comment]
This is blocked until #1982 is merged

**Review by achilleas-k — APPROVED** [review]
LGTM

**Review by msehnout — APPROVED** [review]
Looks nice!

**Pull request merged**
